src/Http/Provider/AbstractHttpProvider.php

```diff
@@ -33,18 +38,14 @@ abstract class AbstractHttpProvider extends AbstractProvider
     private $client;
 
     /**
-     * @var MessageFactory
+     * @var RequestFactoryInterface&StreamFactoryInterface
      */
     private $messageFactory;
 
-    /**
-     * @param ClientInterface     $client
-     * @param MessageFactory|null $factory
-     */
-    public function __construct(ClientInterface $client, MessageFactory $factory = null)
+    public function __construct(ClientInterface $client)
```

Comment:
the removed argument is never used: all implementations override the constructor without passing a second argument ever

Reply:
fixed